integration/flink/src/main/java/io/openlineage/flink/visitor/KafkaSinkVisitor.java

```diff
@@ -46,10 +46,10 @@ public List<OpenLineage.OutputDataset> apply(Object kafkaSink) {
                   datasetFacetsBuilder.schema(
                       AvroSchemaUtils.convert(context.getOpenLineage(), schema)));
 
-      log.debug("Kafka output topic: {}", topic);
+      log.debug("Kafka output topic: {}", di.getName());
 
       return Collections.singletonList(
-          outputDataset().getDataset(topic, bootstrapServers, datasetFacetsBuilder));
+          outputDataset().getDataset(di.getName(), di.getNamespace(), datasetFacetsBuilder));
     } catch (IllegalAccessException e) {
```

Comment:
proposal, add a `getDataset` method that takes in `DatasetIdentifier` instead of passing name and namespace?